src/storage/kv/mod.rs

```diff
@@ -646,10 +646,15 @@ impl<I: Iterator> Cursor<I> {
     pub fn valid(&self) -> Result<bool> {
         if !self.iter.valid() {
             if let Err(e) = self.iter.status() {
-                CRITICAL_ERROR.with_label_values(&["rocksdb"]).inc();
+                CRITICAL_ERROR.with_label_values(&["rocksdb iter"]).inc();
                 if panic_when_unexpected_key_or_data() {
                     set_panic_mark();
                     panic!("Rocksdb error: {}", e);
+                } else {
+                    error!(
+                        "encounter Rocksdb error when iterating";
+                        "err" => %e,
```

Comment:
Prefer "error". Also could you print the current key, `self.min_key` and `self.max_key`?

Reply:
can't print `the current key`, cause cursor already moved on